frontend/src/components/G6Graph.svelte

```diff
@@ -3,40 +3,60 @@
+  import { nonInteractiveState} from '../store';
 
   const dispatch = createEventDispatcher();
   const graph_translation: number = 50;
 
   export let options = {};
   export let data = {};
 
-  let container: HTMLElement;
-  let graph: Graph | null;
-  let timeBarData: TimeBarData[] = [];
+    let nodePoint:number = 0; // variable to retrieve x point of node
 
-  export const reset = () => {
-    graph?.changeData({});
-    timeBarData = [];
-    graph?.render();
-    dispatch("nodeselected", null);
-  };
+    let container: HTMLElement;
+    let graph: Graph | null;
+    let timeBarData: TimeBarData[] = [];
+    export const reset = () => {
+      graph?.changeData({});
+      timeBarData = [];
+      graph?.render();
+      dispatch("nodeselected", null);
+    };
+      // This is a hack to get the graph to render the entire window width
+      export const resizeGraph = () => {
+          if (graph && container) {
+              const width = Number(
+                  window.innerWidth
+              );
+              const height = Number(
+                  window.innerHeight
+              );
+              graph.changeSize(width, height);
+          }
+      };
+    export const focusNode = (id: any) => {
+      graph.focusItem(id);
+    };
 
-    // This is a hack to get the graph to render the entire window width
-  export const resizeGraph = () => {
-        if (graph && container) {
-            const width = Number(
-                window.innerWidth
-            );
-            const height = Number(
-                window.innerHeight
-            );
-            graph.changeSize(width, height);
+     //non-InteractiveMode function to calculate how much translation the graph should make
+    export const nonInteractiveMode = (e:any , modeDisabled:boolean) =>{
+        if(modeDisabled == false){
+            graph.translate(0,0); // to disable nonInteractive mode
         }
-  };
+        else{
+            let oldNodePoint:number = nodePoint; // give initial value of nodepoint to new variable
+            let scrollDistance:number  = 0; // variable to calculate how far apart the two latest nodes on the graph are
+            let scrollDistanceWithRatio:number = 0; // variable to calculate scroll distance variable multiplied with zoom ratio of graph
+            let zoomRatio:number = graph.getZoom(); 
+            nodePoint = e.x - container.scrollWidth;// grab latest node on the graph and find its point
+            if(nodePoint != oldNodePoint){ // find out if latest node has moved to new horizontal position  
+              scrollDistance = nodePoint - oldNodePoint; // if it has moved, find out how far it is from the nearest node to the left
+              scrollDistanceWithRatio = scrollDistance*zoomRatio // multiple with zoom ratio for accurate graph translation
+              graph.translate(-scrollDistanceWithRatio,0); // negate distance to indicate left translation
+            }
+        }
```

Comment:
Code should generally be self documenting, you will not make it far if you need to comment every single line you write hehe; please reduce comments to be supplementary only,

Reply:
I've reduced the comments , let me know if that looks better